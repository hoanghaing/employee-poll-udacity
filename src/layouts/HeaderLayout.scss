.tabs {
  display: inline-block;
  border-bottom: 2px solid rgb(168, 168, 168);
  width: -moz-available;
  margin-left: 2rem;
  margin-right: 2rem;
  .tab {
    margin-top: 0.5rem;
  }
}
.user {
  display: flex;
  margin: 0.5rem auto;
  float: right;
  align-items: center;
  .avt {
    width: 2rem;
    margin-right: 0.5rem;
  }
  .name {
    font-weight: 600;
  }
}
.log-out {
  display: inline;
  margin-left: 1rem;
  margin-top: 0.65rem;
  &:hover {
    cursor: pointer;
    color: rgb(112, 112, 112);
    font-weight: 500;
  }
  float: right;
}