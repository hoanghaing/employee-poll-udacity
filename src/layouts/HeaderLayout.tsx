import React from 'react';
import './HeaderLayout.scss';
import { useNavigate } from "react-router-dom";
const HeaderLayout = ({ children }) => {
  const navigate = useNavigate();
  const onClickLogout = () => {
    localStorage.removeItem("authenticated");
    return navigate("/login");
  }
  return (
    <div>
      <header>
        <div role="tablist" className="tabs">
          <a role="tab" className="tab" href='/home'>Home</a>
          <a role="tab" className="tab" href='/leaderboard'>Leaderboard</a>
          <a role="tab" className="tab" href='/new'>New</a>
          <div className='log-out' onClick={onClickLogout}>
            Logout
          </div>
          <div className='user'>
            <img className='avt' src='./src/assets/avatar/mtsamis.jpg'/>
            <span className='name'>Ahihi</span>
          </div>

        </div>
      </header>
      {children}
    </div>
  );
};

export default HeaderLayout;