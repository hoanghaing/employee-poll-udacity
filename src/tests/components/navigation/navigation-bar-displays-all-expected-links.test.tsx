import { expect, describe, it, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import HeaderLayout from '@/layouts/HeaderLayout';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  user: {
    id: '',
    name: '',
    avatarURL: ''
  }
  // Mock your Redux state here if needed
});

test('Snapshot: navigation bar', async () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <HeaderLayout />
      </BrowserRouter>
    </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
