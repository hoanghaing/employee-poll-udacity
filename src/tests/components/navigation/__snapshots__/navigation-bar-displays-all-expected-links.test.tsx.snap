// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot: navigation bar 1`] = `
<div>
  <header>
    <div
      class="tabs"
      role="tablist"
    >
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/home"
        >
          Home
        </a>
      </span>
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/leaderboard"
        >
          Leaderboard
        </a>
      </span>
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/new"
        >
          New
        </a>
      </span>
      <div
        class="log-out"
      >
        Logout
      </div>
      <div
        class="user"
      >
        <img
          class="avt"
          src="../src/"
        />
        <span
          class="name"
        />
      </div>
    </div>
  </header>
</div>
`;

exports[`renders login form correctly 1`] = `
<div>
  <header>
    <div
      class="tabs"
      role="tablist"
    >
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/home"
        >
          Home
        </a>
      </span>
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/leaderboard"
        >
          Leaderboard
        </a>
      </span>
      <span
        class="tab"
        role="tab"
      >
        <a
          href="/new"
        >
          New
        </a>
      </span>
      <div
        class="log-out"
      >
        Logout
      </div>
      <div
        class="user"
      >
        <img
          class="avt"
          src="../src/"
        />
        <span
          class="name"
        />
      </div>
    </div>
  </header>
</div>
`;
